src/main/java/winstone/Launcher.java

```diff
@@ -152,8 +145,10 @@ public Launcher(Map args) throws IOException {
                 Thread.currentThread().setContextClassLoader( extraClassLoader );
             }
 
-            this.threadPool = createThreadPool();
-            this.server = new Server(new ExecutorThreadPool(threadPool));
+            int qtpMaxThread = Option.QTP_MAXTHREADS.get(args);
+            QueuedThreadPool queuedThreadPool = qtpMaxThread>0?new QueuedThreadPool(qtpMaxThread):new QueuedThreadPool();
```

Comment:
If the option is not specified, then we use an unbounded thread pool. Do we anticipate any problems with that? We could use a default max value instead to avoid unbounded growth.

Reply:
@dwnusbaum see results of scalability testing by @skeenan947 here: https://issues.jenkins-ci.org/browse/JENKINS-33412?focusedCommentId=347435&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-347435

Reply:
@rodrigc Is that the right link? I agree that the change is good and works, I just was curious if there is any benefit to using a max limit to the number of threads even in the default case:

    QueuedThreadPool queuedThreadPool = qtpMaxThread > 0
        ? new QueuedThreadPool(qtpMaxThread)
        : new QueuedThreadPool($SOME_DEFAULT_HERE); // Would it be better to put a (large) max here instead of making it unbounded?

Reply:
default is 200 https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L72  which seems to be a good value.

Reply:
Thanks @olamy looks like I was just confused!